index.bs

```diff
@@ -1450,6 +1451,60 @@ that are returned to the caller when a new credential is created, or a new asser
         [=extension identifier=] → [=client extension output=] entries produced by the extension's
         [=client extension processing=].
 
+    :   {{PublicKeyCredential/toJSON()}}
+    ::  This operation returns {{RegistrationResponseJSON}} or {{AuthenticationResponseJSON}}, that is a JSON-serialized
+        representation of the instance of {{PublicKeyCredential}} suitable for submission to a [=[RP]=] server as a DOMString
+        payload with `application/json` mimetype. The [=client=] is in charge of
+        [serializing values to JSON as usual](https://webidl.spec.whatwg.org/#idl-tojson-operation), but must take additional
+        steps to encode any {{ArrayBuffer}} values to {{DOMString}} values using [=base64url encoding=] to ensure compatibility
+        with JSON syntax. {{ArrayBuffer}} values to be encoded are identified according to the detected type of the instance's
+        `response` property.
+    ::  |clientExtensionResults| below should be the output of {{PublicKeyCredential/getClientExtensionResults()}}, with any
+        {{ArrayBuffer}} values encoded to {{DOMString}} values using [=base64url encoding=]. This may include {{ArrayBuffer}}
+        values from extensions registered in the IANA "WebAuthn Extension Identifiers" registry [[!IANA-WebAuthn-Registries]]
+        but not defined in [[#sctn-extensions]].
```

Comment:
I tried defining this to support any extension that might be returned, but I'm uncertain if clients are expected to be aware of all registered extensions and their various values. What should happen, for example, if a client wasn't aware of `largeBlob` and received its `ArrayBuffer` output? Should it drop the extension value? Return the `ArrayBuffer`?

Reply:
The `largeBlob` extension isn't the best example since there is additional client work needed to create a response.

There are two cases defined for client extension results:

1. An extension understood by the client. This should arguably be the only option, see below
2. An authenticator extension output based on an authenticator extension input that was not understood, but instead processed by the [limited heuristic](https://www.w3.org/TR/webauthn-2/#sctn-extensions). This client extension output might be anything, but is 'suggested' to be a pure JSON version of the authenticator extension output, with no binary conversion allowed.

We may separately want to strike the heuristic for output from level 3. This is not the behavior of the extensions defined after level 1, and AFAIK only `uvm` remains with that behavior defined. Such duplication creates a second source of information for server processing which is not properly protected by the credential or any attestation.